### DISCUSSION TITLE
running targets pipeline using slurm (HPC) AND callr (local)

### DISCUSSION THREAD

**Discussion by wsteenhu** [ropensci/targets#907] [Help] [answered]
I was hoping you could advise on how to organise my pipeline in the following scenario.
I aim to tune multiple machine learning models. While I have an HPC to my disposal, I would like to be able to also run/test the pipeline locally. Also collaborators who would like to run the pipeline themselves should be able to (even if they do not have an HPC).
In my view, this requires defining different future::plan()s in _targets.R, which seems doable (maybe by checking the location [HPC vs local] and setting the plan accordingly). 
More troubling seems to be the scenario where I'd like to set specific resources for specific steps in the pipeline (which may apply to HPC/slurm, but not to locally). 

Any advice on how to go about this is very welcome.

**Comment by wlandau** [comment, marked as answer]
You could program that into `_targets.R`. For the local case, you could set a local `future::plan()` / `clustermq` option, and the opposite for HPC. You could control that either by having your collaborators comment/uncomment certain lines, or you could have an `if`/`else` statement in there. The condition you check could also be linked to target definitions, e.g. `if(run_locally) NULL else tar_target(...)`. All this could happen with custom code.

**Reply by wsteenhu** [reply]
Thank you, this helps a lot.
